src/registrar/models/domain.py

```diff
@@ -304,13 +309,40 @@ def isSubdomain(cls, name: str, nameserver: str):
         regex = re.compile(full_pattern)
         return bool(regex.match(nameserver))
 
+    @classmethod
```

Comment:
**suggestion (blocking)**: This function seems like a perfect candidate for some simple unit tests that check that it performs correctly on various types of input that either conform with the rules or don't. Maybe in `registrar/tests/test_models_domain.py` or `test_models_domain_helper.py`?

Reply:
@neilmb I have added test cases and pushed.  Would you mind taking a look again?